ddl/ddl_db_test.go

```diff
@@ -2444,3 +2448,28 @@ func (s *testDBSuite) TestAlterTableAddPartition(c *C) {
 	);`
 	s.testErrorCode(c, sql7, mysql.ErrSameNamePartition)
 }
+
+func (s *testDBSuite) TestPartitionAddIndex(c *C) {
+	s.tk.MustExec("use test")
+	s.tk.MustExec("set @@tidb_enable_table_partition = 1")
+	s.tk.MustExec(`create table partition_add_idx (
+	id int not null,
+	hired date not null
+	)
+	partition by range( year(hired) ) (
+	partition p1 values less than (1991),
+	partition p3 values less than (2001),
+	partition p3 values less than (2004),
+	partition p3 values less than (2008),
+	partition p3 values less than (2012),
+	partition p3 values less than (2018)
+	);`)
+	for i := 0; i < 500; i++ {
+		s.tk.MustExec(fmt.Sprintf("insert into partition_add_idx values (%d, '%d-01-01')", i, 1988+rand.Intn(30)))
+	}
+
+	s.tk.MustExec("alter table t add index idx (hired)")
+	// TODO: Make admin check table work for partition.
```

Comment:
I'll fix this TODO after this PR is merged https://github.com/pingcap/tidb/pull/6963/files,
because `admin check table` would benefit from that refactor.